src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java

```diff
@@ -263,14 +268,19 @@ private List<List<BuiltCondition>> buildConditions(DBFilterBase.IDFilter additio
     // then add id filter if available, in case of multiple `in` conditions we need to send multiple
     // condtions
     if (idFilterToUse != null) {
-      return idFilterToUse.getAll().stream()
-          .map(
-              idCondition -> {
-                List<BuiltCondition> conditionsWithId = new ArrayList<>(conditions);
-                conditionsWithId.add(idCondition);
-                return conditionsWithId;
-              })
-          .collect(Collectors.toList());
+      final List<BuiltCondition> inSplit = idFilterToUse.getAll();
+      if (inSplit.isEmpty()) {
+        return null;
+      } else {
+        return inSplit.stream()
+            .map(
+                idCondition -> {
+                  List<BuiltCondition> conditionsWithId = new ArrayList<>(conditions);
+                  conditionsWithId.add(idCondition);
+                  return conditionsWithId;
+                })
+            .collect(Collectors.toList());
```

Comment:
Wait so is the primary key supporting the Cassandra `IN` or not? Or we anyway want to run the `IN` as multiple queries?

Reply:
There are 2 reasons in conditions on primary key is run as multiple queries. 
1. Reduce load on Coordinator which eventually does the same thing.
2. Cassandra does not allow additional condition if IN is used on primary key.